### DISCUSSION TITLE
誤変換を集める

### DISCUSSION THREAD

**Discussion by tokuhirom** [akaza-im/akaza#74] [General]
なにか誤変換が起きるケースがあれば集めましょう

**Comment by tokuhirom** [comment]
「あつめましょう」が「集めま賞」になります

**Reply by tokuhirom** [reply]
```
集めましょう
集め    動詞,自立,*,*,一段,連用形,集める,アツメ,アツメ
ましょ  助動詞,*,*,*,特殊・マス,未然ウ接続,ます,マショ,マショ
う      助動詞,*,*,*,不変化型,基本形,う,ウ,ウ
EOS
```

助動詞をまとめて学習しておいたほうが良さそう。「助動詞+助動詞」をまとめて登録すべきか？

**Reply by tokuhirom** [reply]
集め/魔性 になるケースもある。

**Reply by tokuhirom** [reply]
「集めましょう」は複合語化されているのだが、「集めましょう」のコーパス内での頻度が少なすぎるために語彙に登録されていない。どうしたらいいかなぁ。手動で一旦コーパスに登録すれば解決はするんだけど。

**Reply by tokuhirom** [reply]
https://github.com/tokuhirom/akaza/blob/master/akaza-data/corpus/should.txt コーパスに足して解決。。

**Comment by tokuhirom** [comment]
「すんでる」が「既る」になります。既/すんで のコストが安すぎる?

**Reply by tokuhirom** [reply]
`既/すんで       2452` になっていて、妙に頻度が高い。
「既製品」などが「既/すんで 製品/せいひん」になっている。

unidic だと「既製品」は「既製/きせー 品/ひん」 になっていて問題がない。
http://www4414uj.sakura.ne.jp/Yasanichi1/unicheck/

mecab-user-dict.csv に追加が必要かも

**Reply by tokuhirom** [reply]
既に解決済み https://github.com/tokuhirom/akaza/commit/52d927ac8c08e2a0111e66029d2d0c0368c587a7

**Comment by tokuhirom** [comment]
「にながしてる」が「ニナがしてる」になる。「に流してる」になるのが正しいと思う。

> 流して/ながして 2861
> ニナ/にな   170

ニナ、ってなんででてくるんだ。。

**Reply by tokuhirom** [reply]
青空文庫に出てくるキャラクターの名前っぽい。こういうのは除外しないとダメだなぁ。
ただ、こういうのが出てきたとて、`ニナ/がして` より `に/流して` のほうが通りそうなもんなんだよなぁ。
このへんのスコアがどうなってるかの調査をちゃんとすぐにできるようにする必要がある。

**Reply by tokuhirom** [reply]
ニナ、のようなカタカナの2文字のものに関してはノイズになっている可能性が高いので、明示的に除外するほうが良さそう。
ipadic かなにかと突き合わせて、登録されていないものは除外する、とか。
例えば、「タイ」は国名なので登録されていてもいいと思うので。

**Reply by tokuhirom** [reply]
二文字のカタカナを一旦除外するヒューリスティックをいれた

**Comment by tokuhirom** [comment]
- 戦闘文字
- 変換性度も
- 治った仮名

初期にあった誤変換ですがいずれも治っています。

**Comment by tokuhirom** [comment]
変換したい文 → 変換した異文

**Reply by tokuhirom** [reply]
再現せず。

**Comment by tokuhirom** [comment]
「アグリゲーション」が一発で変換できない。

**Reply by tokuhirom** [reply]
カタカナ語は skk 辞書メインだとちょっと弱いのかもしれない。

**Reply by tokuhirom** [reply]
が、ちょうどいいソースが見当たらない。

**Reply by omasanori** [reply]
昔の日本語能力試験の出題基準からカタカナ語リストを抽出している人が。

http://www7a.biglobe.ne.jp/nifongo/data/index.html

**Reply by omasanori** [reply]
Wiktionaryのダンプデータは使い道がありそう。

**Reply by tokuhirom** [reply]
アグリゲーション、は unidic にものっていたので良いかも。
unidic はライセンスが BSD なので。

**Reply by tokuhirom** [reply]
カタカナ語は一般的にSKK辞書には登録されていない。
辞書に登録されていないと、トライに入らないので文節区切りの候補にならないので候補に入れる必要がある。

**Reply by tokuhirom** [reply]
#156 で対応済み。

**Comment by tokuhirom** [comment]
「へん」の候補に「雌鳥王」がでる。なにこれ

**Reply by tokuhirom** [reply]
https://ja.wikipedia.org/wiki/%E3%83%8F%E3%83%BC%E3%83%A9%E3%83%AB3%E4%B8%96_(%E3%83%87%E3%83%B3%E3%83%9E%E3%83%BC%E3%82%AF%E7%8E%8B)
で、`雌鳥王（ヘン[[2]](https://ja.wikipedia.org/wiki/%E3%83%8F%E3%83%BC%E3%83%A9%E3%83%AB3%E4%B8%96_(%E3%83%87%E3%83%B3%E3%83%9E%E3%83%BC%E3%82%AF%E7%8E%8B)#cite_note-Morby97-2)、[デンマーク語](https://ja.wikipedia.org/wiki/%E3%83%87%E3%83%B3%E3%83%9E%E3%83%BC%E3%82%AF%E8%AA%9E): Hen）` という表記があるので、これを拾ってそう。

**Reply by tokuhirom** [reply]
https://github.com/tokuhirom/jawiki-kana-kanji-dict/commit/0750c90f97c743003cc1fff03bc342b3bf5c6009 で対応済み。そのうち取り込まれるでしょう。

**Reply by tokuhirom** [reply]
jawiki-kana-kanji-dict をそもそも読まないことにしたので問題解決した。

**Comment by tokuhirom** [comment]
「だせたらいいよな」が「出せタラ言い寄な」になっている。話し言葉に弱い!

**Reply by tokuhirom** [reply]
```
出せたらいいよな
出せ    動詞,自立,*,*,一段,連用形,出せる,ダセ,ダセ
たら    助動詞,*,*,*,特殊・タ,仮定形,た,タラ,タラ
いい    形容詞,自立,*,*,形容詞・イイ,基本形,いい,イイ,イイ
よ      助詞,終助詞,*,*,*,*,よ,ヨ,ヨ
な      助詞,終助詞,*,*,*,*,な,ナ,ナ
EOS
```

mecab+ipadic ではこうなる。

**Reply by tokuhirom** [reply]
形容詞+助詞(終助詞)+助詞(終助詞) を複合語化した方が良さそう。

**Reply by tokuhirom** [reply]
それはそれとして、一旦コーパスに入れて解決。

**Comment by tokuhirom** [comment]
けんさくしてみて → 検索して見て

これは「みて」を開いた方がいい

**Comment by tokuhirom** [comment]
「ごけんち」が変換できない。
「ご」の候補に豆油、豆汁という謎ワードが存在している。
